### PR TITLE
[index] NFC: Add a symbol kind for init accessor

### PR DESCRIPTION
**Pull request by xedin**
This is a new kind of accessor that is being introduced to the Swift language.

### CODE REVIEW

**Comment by xedin** [comment]
@swift-ci please test

**Comment by xedin** [comment]
@swift-ci please test

File: clang/include/clang/Index/IndexSymbol.h [hunk 1]
```diff
@@ -85,6 +85,7 @@ enum class SymbolSubKind : uint8_t {
 
   SwiftAccessorWillSet,
   SwiftAccessorDidSet,
+  SwiftAccessorInit,
```
Comment (bnbarham) [inline clang/include/clang/Index/IndexSymbol.h]: It'd be better to add this at the end of the enum, even though that makes the ordering a little weird. But better to maintain compatibility IMO. Same with the 5.9.

Comment (bnbarham) [reply clang/include/clang/Index/IndexSymbol.h]: There's also a few other updates to make -
  - `indexstore.h` to add a new `indexstore_symbol_subkind_t` and to bump `INDEXSTORE_VERSION_MINOR`
  - `IndexDataStoreUtils.cpp` for the mapping between `SymbolSubKind` and `indexstore_symbol_subkind_t`
  - `IndexSymbol.cpp` `getSymbolSubKindString`

**Comment by xedin** [comment]
@swift-ci please test

File: clang/include/clang/Index/IndexSymbol.h (hunk 1 above)
Comment (xedin) [reply clang/include/clang/Index/IndexSymbol.h]: Done!

Comment (bnbarham) [reply clang/include/clang/Index/IndexSymbol.h]: Still missing the `getSymbolSubKindString` change in `IndexSymbol.cpp` (I added that to the comment a little after, so if you were looking at the email it probably wasn't in there).

Comment (xedin) [reply clang/include/clang/Index/IndexSymbol.h]: I could swear I added that :/

**Comment by xedin** [comment]
@swift-ci please test

**Review by bnbarham — APPROVED** [review]

**Pull request merged**